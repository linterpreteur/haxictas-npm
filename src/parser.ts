export type MenuData = {
  data: {
      cafeteria: string,
      meals: {[item: string]: string|number}[]
  },
  date: Date
};

export type MenuParams = {data: string, date: Date};

export type CafeteriaData = {
  cafeteria: string,
  location: string,
  hours: {conditions: {day: string[], floor?: string, opens_at: string[], closes_at: string[]}}[]
};

export type CafeteriaParams = string;

export type Parser = {
  menus: (params: MenuParams) => Generator<MenuData, void, undefined>,
  cafeterias: (params: CafeteriaParams) => Generator<CafeteriaData, void, undefined>
}
