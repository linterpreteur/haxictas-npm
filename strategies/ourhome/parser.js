const cheerio = require('cheerio');
const prices = require('./prices');

module.exports.menus = ({data: page}, callback) => {
    const cached = this.cached = this.cached || {};
    const $ = cheerio.load(page);

    const parsePrice = (board) => {
        const price = {};
        const lis = board.find('li');
        if (lis.contents().length) return false;
        lis.each(li => {
            const tag = $(li).attr('class');
            const value = $(li).text();
            if (value.match(/[0-9]원/)) {
                value = parseInt(value.replace(/[^0-9]/, ''), 10);
            }
            price[tag] = value;
        });
        return price;
    };
    
    cached.price = parsePrice($('.board')) || cached.price || prices;
    
    const cafeterias = {};
    $('td[scope=rowgroup]').each((i, td) => {
        cafeterias[$(td).text().trim()] = i;
    });
    
    const dates = [];
    for (let i = 0; i < 7; i++) {
        const date = [];
        for (let j = 0; j < Object.keys(cafeterias).length; j++) {
            date.push({});
        }
        dates.push(date);
    }
    
    let day = 0; // 0 = SUN, 1 = MON, ..., 6 = SAT
    let meal = 0; // 0 = BFST, 1 = LNCH, 2 = DNNR
    let cafeteria = null;
    
    const meals = ['아침', '점심', '저녁'];
    let onRemarksCell;
            
    const isCafeteriaNameCell = (td) => $(td).attr('scope') === 'rowgroup';
    
    $('.t_col td').each((_, td) => {
        const lis = $(td).find('li');
        const cellText = $(td).text().trim();
        
        const cellEmpty = !cellText;
        const cellHasItems = !!lis.contents().length;
        const onTagCell = !cellEmpty && !cellHasItems;
        const onMenuItemCell = cellHasItems || (!onRemarksCell && cellEmpty);
        
        if (onTagCell) {
            onRemarksCell = (cellText === '비고');
            if (isCafeteriaNameCell(td)) cafeteria = cellText.trim();
            if (meals.indexOf(cellText) > -1) meal = meals.indexOf(cellText);
        } else if (onMenuItemCell) {
            const result = dates[day][cafeterias[cafeteria]];
            
            result['cafeteria'] = result['cafeteria'] || cafeteria;
            result['meals'] = result['meals'] || [{}, {}, {}];
            
            lis.each((_, li) => {
                const item = $(li).text().trim();
                const symbol = $(li).attr('class');
                const price = cached.price[symbol] || prices.none;
                
                result['meals'][meal][item] = price;
            });
            
            day = (day + 1) % 7;
        }
    });
    
    dates.forEach((date, day) => {
        date.forEach(cafeteria => {
            callback({day: day, data: cafeteria});
        });
    });
};

module.exports.cafeterias = (page, callback) => {
    const $ = cheerio.load(page);
    const tables = $('.basic_tbl').filter(i => i < 2);
    
    const onBreakPattern = /\(방학중 ([0-9]{2}:[0-9]{2})\)/;
    
    tables.each((i, table) => {
        const result = {
            cafeteria: i === 0 ? '919동' : '아워홈',
            hours: []
        };
        
        $(table).find('tr').each((i, tr) => {
            const tds = $(tr).find('td');
            if (i > 3) {
                result.location = tds.first().text().trim();
            } else {
                const search = function(days) {
                    for (let i = 0; i < result.hours.length; i++) {
                        const target = result.hours[i];
                        if (target.conditions.day.every(x => days.indexOf(x) != -1) &&
                            days.every(x => target.conditions.day.indexOf(x) != -1)) {
                            return target;
                        }
                    }
                    return null;
                };
                
                const update = function(days, hours) {
                    let found = search(days);
                    if (!found) {
                        found = {
                            conditions: {day: days},
                            opens_at: [],
                            closes_at: []
                        };
                        result.hours.push(found);
                    }
                    found.opens_at.push(hours[0].trim());
                    found.closes_at.push(hours[1].trim());
                };
                
                tds.each((i, td) => {
                    let text = $(td).text().trim();
                    const tokens = () => text.split('~');
                    let days = [i === 0 ? 'weekdays' : 'weekends'];
                    
                    if ($(td).attr('colspan') == 2) {
                        days = ['weekdays', 'weekends'];
                    }
                    
                    if (text.indexOf('방학중') != -1) {
                        const onBreakTokens = tokens().map(token => {
                            const match = token.match(onBreakPattern);
                            return (match && match[1]) || token;
                        });
                        
                        update(days.concat('break'), onBreakTokens);
                        
                        text = text.replace(onBreakPattern, '');
                    }
                    update(days, tokens());
                });
            }
        });
        callback(result);
    });
};
