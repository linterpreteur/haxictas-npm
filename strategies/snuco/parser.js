var cheerio = require('cheerio');
var prices = require('./prices');

module.exports.menus = (page, callback) => {
    var cached = this.cached = this.cached || {};
    var $ = cheerio.load(page);
    
    var pricePattern = /[\u24D0-\u24E9] [^ ]+/g;
    var parsePrice = (text) => {
        var price = {};
        var matches = text.match(pricePattern);
        if (!matches) return false;
        matches.forEach((m) => {
            var symbol = m.substring(0, m.indexOf(' '));
            var value = m.substring(m.indexOf(' ') + 1);
            if (value.match(/[0-9]원/)) {
                value = parseInt(value.replace(/[^0-9]/, ''), 10);
            }
            price[symbol] = value;
        });
        return price;
    };
    
    var trs = $('#Content tr');
    
    cached.price = parsePrice(trs.last().find('td').text()) || cached.price || prices;
    
    trs.filter((_, v) => $(v).text().trim()).each((i, tr) => {
        var tds = $(tr).find('td');
        
        var result = {
            cafeteria: null,
            meals: []
        };
        
        var priceSymbols = /[\u24D0-\u24E9]/g;
        var separators = /[/\n]+(?=^|[\u24D0-\u24E9])/;
        
        tds.filter(i => i % 2 === 0).each((i, td) => {
            td = $(td);
            var text = td.text();
            
            if (i === 0) {
                if (!text || text.match(pricePattern)) return false;
                td.html(td.html().replace(/(<.+?>.*)<br>.*(<.+?>)/, '$1$2'));
                result.cafeteria = td.text();
            } else {
                var items = {};
                var matches = text.match(priceSymbols);
                
                if (matches) {
                    text.replace(/\[.+?\]|\(.+?\)/, '')
                        .split(separators).forEach(text => {
                            var match = text.match(priceSymbols);
                            var symbol = match && match[0];
                            var item = text.replace(symbol, '').trim();
                            var price = cached.price[symbol] || prices.none;
                            items[item] = price;
                        });
                } else if (text.trim()) {
                    items[text.trim()] = prices.none;
                }
                result.meals.push(items);
            }
        });
        
        if (!result.cafeteria) return;
        
        callback(result);
    });
};

module.exports.cafeterias = (page, callback) => {
    var $ = cheerio.load(page);
    
    // hard coding hell yeah!
    // have no idea whats going on
    var cafeteria = 'cafeteria';
    var floor = 'floor';
    var days = ['weekdays', 'saturday', 'holidays'];
    var categories = [
        cafeteria,
        'location',
        floor,
        'chairs',
        'target',
        days[0],
        days[1],
        days[2]
    ];
    var secondLineColumnOffset = 2;
    
    var trs = $('#Content tr');
    
    trs.filter((_, v) => $(v).text().trim()).each((i, tr) => {
        var tds = $(tr).find('td');
        var mod = tds.first().text().trim() ? 0 : 1;
        var result = {
            hours: []
        };
        
        var _floor = floor;
        var inserted = {};
        tds.filter(i => i % 2 === mod).each((i, td) => {
            var category = categories[i + (mod === 0 ? 0 : secondLineColumnOffset)];
            td = $(td);
            if (category === cafeteria) {
                td.html(td.html().replace(/(\s+|\n)/g, ' '));
                td.html(td.html().replace(/(<.+?>.*)<br>.*(<.+?>)/, '$1$2'));
            }
            var text = td.text().trim();
            
            if (categories.indexOf(category) < secondLineColumnOffset) {
                result[category] = text;
            } else {
                if (category === floor) {
                    _floor = text;
                } if (days.indexOf(category) != -1) {
                    text = text.replace(/ ?[-~] ?/g, '-');
                    
                    var hour = function(regex) {
                        var matches = text.match(regex);
                        if (!matches) return null;
                        return matches.map(x => x.match(/[0-9:]+/)[0]);
                    };
                    
                    var opens_at = hour(/[0-9]{2}:[0-9]{2}\-/g);
                    var closes_at = hour(/\-[0-9]{2}:[0-9]{2}/g);
                    
                    var dupe = inserted[_floor + text];
                    if (isFinite(dupe)) {
                        result.hours[dupe].conditions.day.push(category);
                    } else if (opens_at || closes_at) {
                        result.hours.push({
                            conditions: {
                                floor: _floor,
                                day: [category]
                            },
                            opens_at: opens_at,
                            closes_at: closes_at
                        });
                        inserted[_floor + text] = result.hours.length - 1;
                    }
                }
            }
        });
        callback(result);
    });
};