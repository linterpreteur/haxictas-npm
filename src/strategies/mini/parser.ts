import {JSDOM} from 'jsdom';
import {MenuParams, MenuCallback} from '../../parser';
import {dedupe, querySelectorArray, normalize} from '../../lib/utils';

export function menus({data, date}: MenuParams, callback: MenuCallback) {
    const {document} = (new JSDOM(data)).window;

    const cafeterias = dedupe(
        querySelectorArray(document, 'td.head')
            .map((x) => normalize(x.textContent))
    );

    const menu: {[cafeteria: string]: {[item: string]: number}[]} = {};
    cafeterias.forEach(cafeteria => { menu[cafeteria] = [] });
    
    const cells = querySelectorArray(document, 'td');
    let meal = -1;
    let cafeteria = null;
    cells.forEach(cell => {
        const text = cell.textContent.trim();

        switch (text) {
            case '아침':
                meal = 0;
                return;
            case '점심':
                meal = 1;
                return;
            case '저녁':
                meal = 2;
                return;
        }

        const priceClass = 'price';
        const priceSelector = `.${priceClass}`;
        const isMenuItem = !!cell.querySelector(priceSelector);
        if (!isMenuItem) {
            cafeteria = text;
            return;
        }

        let price: number;
        cell.childNodes.forEach(child => {
            if (child instanceof child.ownerDocument.defaultView['Element']) {
                const isPrice = (child as Element).classList.contains(priceClass);
                if (!isPrice) {
                    return;
                }
                const priceSymbol = normalize(child.textContent);
                price = parseInt(priceSymbol, 10) * 100;
                return;
            }
            const item = normalize(child.textContent);
            menu[cafeteria][meal] = Object.assign(
                {},
                (menu[cafeteria][meal] || {}),
                { [item]: price }
            );
        });
    });

    Object.keys(menu).forEach(cafeteria => {
        const meals = menu[cafeteria]
        const data = {
            cafeteria: cafeteria,
            meals: meals
        }
        callback({
            data: data,
            date: date
        })
    })
};
