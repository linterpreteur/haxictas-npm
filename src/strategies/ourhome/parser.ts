import * as cheerio from 'cheerio';
import prices from './prices';
import {MenuParams, MenuCallback, CafeteriaParams, CafeteriaCallback, CafeteriaData} from '../../parser';

export function menus({data: page, date: startDate}: MenuParams, callback: MenuCallback) {
    const cached = this.cached = this.cached || {};
    const $ = cheerio.load(page);

    const parsePrice = (board: Cheerio) => {
        const price = {};
        const lis = board.find('li');
        if (lis.contents().length) return false;
        lis.each(li => {
            const tag = $(li).attr('class');
            let value = $(li).text();
            if (value.match(/[0-9]원/)) {
                value = value.replace(/[^0-9]/, '');
            }
            price[tag] = value;
        });
        return price;
    };
    
    cached.price = parsePrice($('.board')) || cached.price || prices;
    
    const cafeterias = {};
    $('td[scope=rowgroup]').each((i, td) => {
        cafeterias[$(td).text().trim()] = i;
    });
    
    const dates: {cafeteria: string, meals: {[item: string]: number}[]}[][] = [];
    for (let i = 0; i < 7; i++) {
        const date = [];
        for (let j = 0; j < Object.keys(cafeterias).length; j++) {
            date.push({});
        }
        dates.push(date);
    }
    
    let day = 0; // 0 = SUN, 1 = MON, ..., 6 = SAT
    let meal = 0; // 0 = BFST, 1 = LNCH, 2 = DNNR
    let cafeteria = null;
    
    const meals = ['아침', '점심', '저녁'];
    let onRemarksCell: boolean;
            
    const isCafeteriaNameCell = (td) => $(td).attr('scope') === 'rowgroup';
    
    $('.t_col td').each((_, td) => {
        const lis = $(td).find('li');
        const cellText = $(td).text().trim();
        
        const cellEmpty = !cellText;
        const cellHasItems = !!lis.contents().length;
        const onTagCell = !cellEmpty && !cellHasItems;
        const onMenuItemCell = cellHasItems || (!onRemarksCell && cellEmpty);
        
        if (onTagCell) {
            onRemarksCell = (cellText === '비고');
            if (isCafeteriaNameCell(td)) cafeteria = cellText.trim();
            if (meals.indexOf(cellText) > -1) meal = meals.indexOf(cellText);
        } else if (onMenuItemCell) {
            const result = dates[day][cafeterias[cafeteria]];
            
            result['cafeteria'] = result['cafeteria'] || cafeteria;
            result['meals'] = result['meals'] || [{}, {}, {}];
            
            lis.each((_, li) => {
                const item = $(li).text().trim();
                const symbol = $(li).attr('class');
                const price = cached.price[symbol] || prices.none;
                
                result['meals'][meal][item] = price;
            });
            
            day = (day + 1) % 7;
        }
    });
    
    dates.forEach((date, day) => {
        date.forEach(cafeteria => {
            const menuDate = new Date(startDate);
            menuDate.setDate(startDate.getDate() + (day - startDate.getDay()) % 7);
            callback({date: menuDate, data: cafeteria});
        });
    });
};

export function cafeterias(page: CafeteriaParams, callback: CafeteriaCallback) {
    const $ = cheerio.load(page);
    const tables = $('.basic_tbl').filter(i => i < 2);
    
    const onBreakPattern = /\(방학중 ([0-9]{2}:[0-9]{2})\)/;
    
    tables.each((i, table) => {
        const result: CafeteriaData = {
            cafeteria: i === 0 ? '919동' : '아워홈',
            hours: [],
            location: undefined
        };
        
        $(table).find('tr').each((i, tr) => {
            const tds = $(tr).find('td');
            if (i > 3) {
                result.location = tds.first().text().trim();
            } else {
                const search = function(days: string[]) {
                    for (let i = 0; i < result.hours.length; i++) {
                        const target = result.hours[i];
                        if (target.conditions.day.every(x => days.indexOf(x) != -1) &&
                            days.every(x => target.conditions.day.indexOf(x) != -1)) {
                            return target;
                        }
                    }
                    return null;
                };
                
                const update = function(days: string[], hours: string[]) {
                    let found = search(days);
                    if (!found) {
                        found = {
                            conditions: {
                                day: days,
                                opens_at: [],
                                closes_at: []
                            },
                        };
                        result.hours.push(found);
                    }
                    found.conditions.opens_at.push(hours[0].trim());
                    found.conditions.closes_at.push(hours[1].trim());
                };
                
                tds.each((i, td) => {
                    let text = $(td).text().trim();
                    const tokens = () => text.split('~');
                    let days = [i === 0 ? 'weekdays' : 'weekends'];
                    
                    if (parseInt($(td).attr('colspan'), 10) === 2) {
                        days = ['weekdays', 'weekends'];
                    }
                    
                    if (text.indexOf('방학중') != -1) {
                        const onBreakTokens = tokens().map(token => {
                            const match = token.match(onBreakPattern);
                            return (match && match[1]) || token;
                        });
                        
                        update(days.concat('break'), onBreakTokens);
                        
                        text = text.replace(onBreakPattern, '');
                    }
                    update(days, tokens());
                });
            }
        });
        callback(result);
    });
};
