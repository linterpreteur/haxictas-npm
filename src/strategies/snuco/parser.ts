import {JSDOM} from 'jsdom';
import {normalize, querySelectorArray} from '../../lib/utils';
import {MenuParams, MenuCallback, CafeteriaParams, CafeteriaCallback, CafeteriaData} from '../../parser';

export function menus({data: page, date}: MenuParams, callback: MenuCallback) {
    const {document} = (new JSDOM(page)).window;

    function parseMenu(s: string): {[item: string]: number} {
        return s.split(/\n+/g)
            .map(x => x.match(/(.+?)\s*([\d,.]+원)$/) || x)
            .map(x => (typeof x === 'string') ? [x, '기타'] : [x[1], x[2]])
            .map(([k, v]) => [normalize(k), normalize(v)])
            .filter(x => x && x[0])
            .reduce((acc, [k, v]) => Object.assign({}, acc, {[k]: v}), {});
    }
    querySelectorArray(document, 'tbody tr').map(row => {
        const [cafeteria, breakfast, lunch, dinner] = querySelectorArray(row, 'td');
        const data = {
            cafeteria: normalize(cafeteria.textContent),
            meals: [
                parseMenu(breakfast.textContent),
                parseMenu(lunch.textContent),
                parseMenu(dinner.textContent)
            ]
        };
        callback({data: data, date: date});
    });
};

export function cafeterias(page: CafeteriaParams, callback: CafeteriaCallback) {
    const {document} = (new JSDOM(page)).window;

    const header = querySelectorArray(document, 'thead th:not([colspan])');
    const nameIndex = 0;
    const locationIndex = 1;

    const info: CafeteriaData = {
        cafeteria: undefined,
        location: undefined,
        hours: undefined
    };
    let cafeteria = null;
    let floor = null;
    querySelectorArray(document, 'tbody tr').forEach(row => {
        const cells = querySelectorArray(row, 'td');

        const isFullRow = cells.length >= header.length;
        if (isFullRow) {
            const name = cells[nameIndex];
            cafeteria = normalize(name.textContent);

            const location = cells[locationIndex];
            info[cafeteria] = Object.assign(
                {},
                info[cafeteria] || {},
                { location: normalize(location.textContent) }
            );

            cells.splice(cells.findIndex(x => x === name), 1);
            cells.splice(cells.findIndex(x => x === location), 1);
        }

        const mayContainFloor = cells.length > 5;
        if (mayContainFloor) {
            floor = normalize(cells[0].textContent);
            cells.splice(0, 1);
        }
        
        const [size, _, weekdays, saturday, holidays] = cells;

        const specialTypes = /(채식)/;
        const specialTypeMatches = size.textContent.match(specialTypes);
        if (specialTypeMatches) {
            floor = normalize(specialTypeMatches[1]);
        }

        function parseHours(s) {
            const numeralHours = /(\d+:\d+)-(\d+:\d+)/;
            const specialHours = /분식/;
            if (!s.match(numeralHours)) {
                return [normalize(s)];
            }
            if (s.match(specialHours)) {
                return [normalize(s)];
            }
            return s.split(/\n+/)
                .map(normalize)
                .map(x => x.match(numeralHours))
                .filter(x => x)
                .map(x => [x[1], x[2]]);
        }
        const hours = {
            weekdays: parseHours(weekdays.textContent),
            saturday: parseHours(saturday.textContent),
            holidays: parseHours(holidays.textContent)
        };

        function isEqualEntry(a, b) {
            return a.floor === b.floor &&
                a['opens_at'] === b['opens_at'] &&
                a['closes_at'] === b['closes_at'];
        }
        const aggregatedHours = Object.keys(hours)
            .reduce((acc, day) => {
                const newEntry = hours[day].map(item => ({
                    day: [day],
                    floor: floor,
                    'opens_at': (typeof item === 'string') ? item : item[0],
                    'closes_at': (typeof item === 'string') ? item : item[1],
                }));
                return acc.concat(newEntry);
            }, [])
            .reduce((acc, x) => {
                const equalEntry = acc.find(y => isEqualEntry(x, y));
                if (!equalEntry) {
                    return acc.concat(x);
                }
                equalEntry.day.push(...x.day);
                return acc;
            }, [])
            .map(x => ({conditions: x}));
        info[cafeteria] = Object.assign(
            {},
            info[cafeteria] || {},
            { hours: aggregatedHours }
        );
    });
    callback(info);
};
