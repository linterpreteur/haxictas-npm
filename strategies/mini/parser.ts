import {JSDOM} from 'jsdom';
import {MenuParams, MenuCallback} from '../../parser';
import {dedupe, querySelectorArray} from '../../lib/utils';

export function menus({data, date}: MenuParams, callback: MenuCallback) {
    const {document} = (new JSDOM(data)).window;

    const cafeterias = dedupe(
        querySelectorArray(document, 'td.head')
            .map((x) => x.textContent.replace(/\n/g, '').trim())
    );

    const menu: {[cafeteria: string]: {[item: string]: number}[]} = {};
    cafeterias.forEach(cafeteria => { menu[cafeteria] = [] });
    
    const cells = querySelectorArray(document, 'td');
    let meal = -1;
    let cafeteria = null;
    cells.forEach(cell => {
        const text = cell.textContent.trim();

        switch (text) {
            case '아침':
                meal = 0;
                return;
            case '점심':
                meal = 1;
                return;
            case '저녁':
                meal = 2;
                return;
        }

        const isMenuItem = text.match(/^\d+ /)
        if (!isMenuItem) {
            cafeteria = text;
            return;
        }

        const priceSelector = '.price';
        const priceSymbol = cell.querySelector(priceSelector).textContent.trim();
        const price = parseInt(priceSymbol, 10) * 100;
        const item = (() => {
            const tmp = cell.cloneNode(true) as typeof cell;
            const priceNode = tmp.querySelector(priceSelector);
            tmp.removeChild(priceNode);
            return tmp.textContent.replace(/\s{2,}/g, ' ').trim();
        })();
        menu[cafeteria][meal] = Object.assign(
            {},
            (menu[cafeteria][meal] || {}),
            { [item]: price }
        )
    });

    Object.keys(menu).forEach(cafeteria => {
        const meals = menu[cafeteria]
        const data = {
            cafeteria: cafeteria,
            meals: meals
        }
        callback({
            data: data,
            date: date
        })
    })
};

export function cafeterias() { /* no-op */ };
