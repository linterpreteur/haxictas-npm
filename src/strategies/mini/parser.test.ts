import {mini as html} from '../../test/data';
import {menus} from './parser';

describe(menus, () => {
  it('parses given page', () => {
    const params = {data: html, date: new Date(0)};

    menus(params, (x, err) => {

      expect(err).not.toBe(expect.anything());
      expect(x.data).toMatchSnapshot();
    });
  });
});
