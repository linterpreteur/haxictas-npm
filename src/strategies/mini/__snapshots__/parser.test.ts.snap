// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menus parses given page 1`] = `
Object {
  "cafeteria": "학생회관",
  "meals": Array [],
}
`;

exports[`menus parses given page 2`] = `
Object {
  "cafeteria": "기숙사(919동)",
  "meals": Array [],
}
`;

exports[`menus parses given page 3`] = `
Object {
  "cafeteria": "기숙사(901동)",
  "meals": Array [
    Object {
      "계란버터비빔밥/맛살야채볶음": 3000,
      "토스트,씨리얼,우유": 2500,
    },
    Object {
      "불고기버섯전골": 4000,
      "치킨마요덮밥": 4000,
      "탄탄멘": 3500,
    },
    Object {
      "고깃집돌솥밥/두부구이": 4000,
      "짜계치/교자만두": 4000,
    },
  ],
}
`;

exports[`menus parses given page 4`] = `
Object {
  "cafeteria": "전망대(농대)",
  "meals": Array [
    undefined,
    Object {
      "갈비찜": 4000,
      "새우커리": 3500,
    },
  ],
}
`;

exports[`menus parses given page 5`] = `
Object {
  "cafeteria": "두레미담",
  "meals": Array [
    undefined,
    Object {
      "잡곡밥,돈육김치찌개,안동찜닭,오징어링튀김,청포묵김무침,건새우풋고추조림,깍두기,그린샐러드,오늘의차": 6000,
    },
    Object {
      "잡곡밥,건새우아욱국,매콤닭볶음탕,짜장떡볶이,온두부찜*양념간장,흑임자숙주나물,김치,그린샐러드,오늘의차": 6000,
    },
  ],
}
`;

exports[`menus parses given page 6`] = `
Object {
  "cafeteria": "서당골(사범대)",
  "meals": Array [
    undefined,
    Object {
      "굴소스햄야채볶음밥&초코와플": 4000,
      "바몬드카레라이스&비빔만두": 4500,
    },
    Object {
      "하와이안함박스테이크&계란f": 4500,
    },
  ],
}
`;

exports[`menus parses given page 7`] = `
Object {
  "cafeteria": "아름드리(예술)",
  "meals": Array [
    undefined,
    Object {
      "라면": 3000,
      "함박스테이크&갈릭소스": 4500,
      "해물볶음우동&순살치킨가스": 5500,
    },
    Object {
      "스팸김치볶음밥&달걀후라이": 4500,
    },
  ],
}
`;

exports[`menus parses given page 8`] = `
Object {
  "cafeteria": "감골식당",
  "meals": Array [
    undefined,
    Object {
      "김치나베돈가스+미니우동": 4000,
      "청국장찌개+미트볼조림": 3000,
      "콩고기야채불고기,마파두부,쫄면채소무침,연근모듬콩조림,깻잎쌈&당근스틱,유부호박된장국,옥수수죽,백김치,diy샐러드,과일,음료🥗": 6500,
    },
    Object {
      "마라치킨마요덮밥": 4000,
    },
  ],
}
`;

exports[`menus parses given page 9`] = `
Object {
  "cafeteria": "동원관",
  "meals": Array [],
}
`;

exports[`menus parses given page 10`] = `
Object {
  "cafeteria": "자하연",
  "meals": Array [],
}
`;

exports[`menus parses given page 11`] = `
Object {
  "cafeteria": "220동",
  "meals": Array [
    undefined,
    Object {
      "[뚝]치즈미트스파게티*해시브라운": 4500,
      "설렁탕&계란장조림": 4000,
    },
    Object {
      "돈가스덮밥&갈비만두찜": 4500,
      "비빔칼국수&갈비만두찜": 4000,
    },
  ],
}
`;

exports[`menus parses given page 12`] = `
Object {
  "cafeteria": "301동(교수)",
  "meals": Array [
    undefined,
    Object {
      "닭갈비덮밥,사각어묵국": 6000,
      "사골곰탕,파래김구이&양념장": 5500,
    },
  ],
}
`;

exports[`menus parses given page 13`] = `
Object {
  "cafeteria": "302동",
  "meals": Array [],
}
`;

exports[`menus parses given page 14`] = `
Object {
  "cafeteria": "301동",
  "meals": Array [
    undefined,
    undefined,
    Object {
      "고기소보로덮밥": 3500,
      "낙지돌솥비빔밥": 4500,
      "돈육간장덮밥": 4300,
      "등심돈가스": 4000,
      "뚝배기불고기": 4800,
      "유니짜장면": 3500,
      "차돌짬뽕": 5500,
      "치킨스테이크": 4800,
    },
  ],
}
`;

exports[`menus parses given page 15`] = `
Object {
  "cafeteria": "공깡",
  "meals": Array [
    undefined,
    undefined,
    Object {
      "고기 소보로 볶음밥": 3500,
      "공간 짜장면": 3500,
      "공간 짬뽕/밥": 3500,
      "공간 짬짜면": 4500,
      "돈까스 비빔면": 5200,
      "부대짬뽕": 5200,
      "사천 짜장면": 3500,
      "순두부 만두 짬뽕": 4800,
      "유린 만두": 2000,
      "짜장면/짬뽕 곱빼기": 4000,
      "짬짜탕면": 6500,
      "치킨탕수육": 7000,
    },
  ],
}
`;

exports[`menus parses given page 16`] = `
Object {
  "cafeteria": "두레미담(주문식)",
  "meals": Array [
    undefined,
    undefined,
    Object {
      "7선산채비빔밥": 13000,
      "고등어구이정식": 11000,
      "고등어무조림정식": 10000,
      "낙지순두부": 10000,
      "능이버섯삼계탕": 16000,
      "들깨칼국수&왕만두": 10000,
      "묵은지김치찌개": 8000,
      "미담삼계탕": 13000,
      "삼겹보쌈정식": 13000,
      "장어한마리덮밥정식": 15000,
      "전복삼계탕": 19500,
      "전복한우등심샤브샤브": 25000,
      "전복해물된장찌개": 12000,
      "차돌된장찌개": 8000,
      "철판제육볶음정식": 13000,
      "철판주꾸미볶음정식": 13000,
      "초계국수&왕만두": 10000,
      "한우등심샤브샤브": 19000,
      "한우불고기정식": 17000,
      "해물파전": 15000,
      "황태구이정식": 12000,
    },
  ],
}
`;
